modules/ding_webtrekk/ding_webtrekk.module

```diff
@@ -90,35 +153,170 @@ function ding_webtrekk_views_post_execute(&$view) {
     $loaded = TRUE;
 
     $params = [
-      'internalSearch' => array_shift($view->args),
-      'numberSearchResults' => $view->total_rows,
-      'pageTitle' => 'Search content',
+      'OSS' => array_shift($view->args),
+      'OSSr' => $view->total_rows,
     ];
 
-    _ding_webtrekk_set_data_layer_values($params);
+    // If using views pager; track the current search result page here also.
+    if (isset($view->query->pager->current_page)) {
+      $params['p_s_Page'] = ++$view->query->pager->current_page;
+    }
+
+    ding_webtrekk_add_page_parameters($params);
   }
 }
 
 /**
- * Pass Tag Information variables to the inline js script.
+ * Implements hook_ding_entity_view().
+ */
+function ding_webtrekk_ding_entity_view($entity, $view_mode) {
+  $entity_path = entity_uri('ting_object', $entity)['path'];
+
+  // The page parametes for the ting object view should only be added on the
+  // full page view for the material.
+  if ($view_mode == 'full' && current_path() == $entity_path) {
+    $params = [
+      'p_mat_type' => $entity->getType(),
+      'p_mat_indexno' => $entity->getClassification(),
+      'p_mat_lang' => $entity->getLanguage(),
+      'p_mat_source' => $entity->getAc_source(),
+      'p_mat_category' => implode(';', $entity->reply->getGenre()),
+    ];
+    ding_webtrekk_add_page_parameters($params);
+  }
+
+  // Look for at reserve button on this ding entity and setup a Webtrekk event
+  // if present. These could be added in every view_mode and page.
+  if (isset($entity->content['ding_entity_buttons'])) {
```

Comment:
Hmmm... a case for a `ding_entity_buttons_alter` 🤔 .

Reply:
Maybe, I'll check it out..